File: leetcode_solution/Remove Element.py
class Solution:
    # @param    A       a list of integers
    # @param    elem    an integer, value need to be removed
    # @return an integer
    def removeElement(self, A, elem):
        index = 0
        for i in range(0,len(A)):
            if A[i] != elem:
                #A.pop([i]) 
                A[index] = A[i]# without becomes wrong answer. [4,5]4input, output[4] instead of [5]
                index += 1
                
        return index

File: leetcode_java/Remove Element.java
public class Solution {
    public int removeElement(int[] A, int elem) {
        int loopPointer = 0;
        int newLengthPointer = 0;

        while (loopPointer < A.length) {
        	if(A[loopPointer] != elem) {
        		A[newLengthPointer] = A[loopPointer];
        		newLengthPointer ++;
        	}
        	loopPointer ++;
        }
        return newLengthPointer;
    }
}